hircluster.h

```diff
@@ -132,6 +132,8 @@ typedef struct redisClusterContext {
     void *ssl; /* Pointer to a redisSSLContext when using SSL/TLS. */
     sslInitFn *ssl_init_fn; /* Func ptr for SSL context initiation */
 
+    void (*on_connect)(const struct redisContext *c, int status);
```

Comment:
There are typedefs for other callbacks, mostly in the async-api but also for ssl initiation in hiredis-cluster (and hiredis). Maybe we should use one here as well.
..
I now see that there might be a problem with naming since where already is an `redisConnectCallback`.. I guess this is a reason

Reply:
Yes, the naming is unfortunate. :-) That's why I didn't add one. We can call it redisConnectSyncCallback or something like that, if you think it's necessary. I don't think it's necessary since the user defines the function and passes it, but typically doesn't need to use its function type.